crates/cli/src/templates/pom.xml

```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+    xmlns="http://maven.apache.org/POM/4.0.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0
+                        http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>PARSER_NS</groupId>
+  <artifactId>jtreesitter-KEBAB_PARSER_NAME</artifactId>
+  <name>JTreeSitter CAMEL_PARSER_NAME</name>
+  <version>PARSER_VERSION</version>
+  <description>PARSER_DESCRIPTION</description>
+  <url>PARSER_URL</url>
+  <licenses>
+    <license>
+      <name>PARSER_LICENSE</name>
+      <url>https://spdx.org/licenses/PARSER_LICENSE.html</url>
+    </license>
+  </licenses>
+  <developers>
+    <developer>
+      <name>PARSER_AUTHOR_NAME</name>
+      <email>PARSER_AUTHOR_EMAIL</email>
+      <url>PARSER_AUTHOR_URL</url>
+    </developer>
+  </developers>
+  <scm>
+    <url>PARSER_URL</url>
+    <connection>scm:git:git://PARSER_URL_STRIPPED.git</connection>
+    <developerConnection>scm:git:ssh://PARSER_URL_STRIPPED.git</developerConnection>
+  </scm>
+  <properties>
+    <maven.compiler.release>22</maven.compiler.release>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.deploy.skip>true</maven.deploy.skip>
+  </properties>
+  <dependencies>
+    <dependency>
+      <groupId>io.github.tree-sitter</groupId>
+      <artifactId>jtreesitter</artifactId>
+      <version>0.25.4</version>
+      <optional>true</optional>
```

Comment:
Why `optional`? jtreesitter is required to use the bindings.

If a consuming project needs a different jtreesitter version, then it can still override the version (respectively the user can just change it in this `pom.xml` file).
Making it `optional` requires that consuming projects always have to specify a jtreesitter dependency, which might make usage more cumbersome.

Reply:
The tree-sitter dependency is optional in all of the bindings.
This makes it possible to use a hypothetical fork instead.

Reply:
Ah ok. But do you think that this case will be common? I assume it is the exception, and in that case users might rather edit the `pom.xml` directly. Otherwise they risk that their fork causes issues for the bindings, but would not notice it for the bindings themselves (since those use the official jtreesitter) and only notice it for consuming projects.